Loop/Managers/DeviceDataManager.swift

```diff
@@ -181,16 +201,33 @@ class DeviceDataManager: CarbStoreDelegate, TransmitterDelegate, ReceiverDelegat
         status.glucoseDateComponents?.timeZone = pumpState?.timeZone
 
         // The pump sends the same message 3x, so ignore it if we've already seen it.
-        guard status != latestPumpStatus, let pumpDate = status.pumpDateComponents.date else {
+        guard status != latestPumpStatusFromMySentry, let pumpDate = status.pumpDateComponents.date else {
             return
         }
 
         // Report battery changes to Analytics
-        if let latestPumpStatus = latestPumpStatus where status.batteryRemainingPercent - latestPumpStatus.batteryRemainingPercent >= 50 {
+        if let latestPumpStatusFromMySentry = latestPumpStatusFromMySentry where status.batteryRemainingPercent - latestPumpStatusFromMySentry.batteryRemainingPercent >= 50 {
             AnalyticsManager.sharedManager.pumpBatteryWasReplaced()
         }
 
-        latestPumpStatus = status
+        latestPumpStatusFromMySentry = status
+
+
+        // Gather PumpStatus from MySentry packet
+        let pumpStatus: NightscoutUploadKit.PumpStatus?
+        if let pumpDate = status.pumpDateComponents.date, let pumpID = pumpID {
+
+            let batteryStatus = BatteryStatus(percent: status.batteryRemainingPercent, status: "normal")
```

Comment:
Should `"normal"` be a default value in the `BatteryStatus` constructor instead?